tests/conftest.py

```diff
@@ -269,12 +271,12 @@ def factory(
                 wait=False
             )
             assert pvc_obj, "Failed to create PVC"
+            instances.append(pvc_obj)
+            pvc_obj.storageclass = storageclass
```

Comment:
I would keep this on its original place so it can create reference if object is provided in parameter. It provides consistency for calling the factory. Also if we want to remove this reference for custom_data it should be done in all fixtures.

Reply:
I'm not sure it ever worked like that cause if you providing custom_data, you expect that all information would be there and therefore, you don't have storageclass initialized or project.

Is that correct?

Reply:
You can have reference to name of some object in custom_data. You may want to keep the reference to object with that name also in object using it. However, it's not mandatory. If you don't provide parameter with the reference then the reference to object is empty.

If you want to change this functionality then I propose to do it in different PR as this PR is about a new fixture. Also it would need to be changed like this in all fixtures.

Reply:
ok reverting this change and let's discuss this in new PR